packages/jupyter-ai/jupyter_ai/mcp/mcp_config_loader.py

```diff
@@ -0,0 +1,107 @@
+import json
+from pathlib import Path
+from typing import Any
+
+from jsonschema import ValidationError, validate
+from jsonschema.exceptions import SchemaError
+
+
+class MCPConfigLoader:
+    """Loader for MCP server configuration files with JSON schema validation."""
+
+    def __init__(self):
+        # Load the schema from the schema.json file
+        schema_path = Path(__file__).parent / "schema.json"
```

Comment:
[minor] I recommend clearly marking this as a top-level constant defined outside the class:

```suggestion
MCP_CONFIG_SCHEMA = Path(__file__).parent / "schema.json"
```